Neos.ContentRepository.Core/Classes/ContentRepository.php

```diff
@@ -101,38 +100,64 @@ public function handle(CommandInterface $command): void
     {
         // the commands only calculate which events they want to have published, but do not do the
         // publishing themselves
-        $eventsToPublish = $this->commandBus->handle($command, $this->commandHandlingDependencies);
-
-        // TODO meaningful exception message
-        $initiatingUserId = $this->userIdProvider->getUserId();
-        $initiatingTimestamp = $this->clock->now()->format(\DateTimeInterface::ATOM);
-
-        // Add "initiatingUserId" and "initiatingTimestamp" metadata to all events.
-        //                        This is done in order to keep information about the _original_ metadata when an
-        //                        event is re-applied during publishing/rebasing
-        // "initiatingUserId": The identifier of the user that originally triggered this event. This will never
-        //                     be overridden if it is set once.
-        // "initiatingTimestamp": The timestamp of the original event. The "recordedAt" timestamp will always be
-        //                        re-created and reflects the time an event was actually persisted in a stream,
-        // the "initiatingTimestamp" will be kept and is never overridden again.
-        // TODO: cleanup
-        $eventsToPublish = new EventsToPublish(
-            $eventsToPublish->streamName,
-            Events::fromArray(
-                $eventsToPublish->events->map(function (EventInterface|DecoratedEvent $event) use (
-                    $initiatingUserId,
-                    $initiatingTimestamp
-                ) {
-                    $metadata = $event instanceof DecoratedEvent ? $event->eventMetadata?->value ?? [] : [];
-                    $metadata['initiatingUserId'] ??= $initiatingUserId;
-                    $metadata['initiatingTimestamp'] ??= $initiatingTimestamp;
-                    return DecoratedEvent::create($event, metadata: EventMetadata::fromArray($metadata));
-                })
-            ),
-            $eventsToPublish->expectedVersion,
-        );
-
-        $this->eventPersister->publishEvents($this, $eventsToPublish);
+        $eventsToPublishOrGenerator = $this->commandBus->handle($command, $this->commandHandlingDependencies);
+
+        // TODO only apply metadata if is copied event?? In generator below we ignore this...
```

Comment:
my bad: we already only set metadata if they didn't exist so we can safely do this in both cases.

We should extract the common code to a method